### PR TITLE
GitHub actions poetry

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6069

## Description
Switched to poetry instead of pipenv in some github actions.

### CODE REVIEW

File: .github/workflows/check-contribution-form-filled.yml [hunk 1]
```diff
@@ -18,17 +18,21 @@ jobs:
...
         run: |
-          python -m pip install --upgrade pip
-          pip install pipenv==2021.5.29
-          cd Utils/github_workflow_scripts
-          pipenv sync
+#          python -m pip install --upgrade pip
+#          pip install pipenv==2021.5.29
+#          cd Utils/github_workflow_scripts
+#          pipenv sync
+          # should i keep the cd Utils/github_workflow_scripts??
```
Comment (dorschw) [inline .github/workflows/check-contribution-form-filled.yml:29]: if everything works without it, then no need

File: Utils/github_workflow_scripts/Pipfile [hunk 2]
```diff
@@ -4,16 +4,16 @@ url = "https://pypi.org/simple"
...
-slack_sdk = "*"
-GitPython = "*"
-demisto-sdk = "*"
+pygithub = "*" # done in ci
+requests = "*" # in dev group, to add to ci?
+blessings = "*" # done in ci
+dateparser = "*" # added to ci
+slack_sdk = "*" # in ci
+GitPython = "*" # Aadded to ci
+demisto-sdk = "*"   # added to ci, but demisto_sdk in dev, same?
```
Comment (dorschw) [inline Utils/github_workflow_scripts/Pipfile:16]: @ilaner do we want to / can we take it from content master somehow?

File: pyproject.toml [hunk 3]
```diff
@@ -13,6 +13,7 @@ defusedxml = "^0.7.1"
 demisto_sdk = "1.16.0"
 requests = "^2.22.0"
 pre-commit = "^2.20.0"
+flake8 = "^4.0.1"
```
Comment (dorschw) [inline pyproject.toml:16]: why?

File: pyproject.toml [hunk 4]
```diff
@@ -123,6 +124,9 @@ google-cloud-bigquery = "^3.2.0"
 db-dtypes = "^1.0.1"
 google-cloud-secret-manager = "^2.13.0"
 json5 = "^0.9.10"
+dataparser = "*"
+GitPython = "*"
+demisto-sdk = "*"
```
Comment (dorschw) [inline pyproject.toml:129]: don't do that, we have a locked version for a reason.

**Review by dorschw — CHANGES_REQUESTED** [review]

**Review by ilaner — APPROVED** [review]
Good job!

File: pyproject.toml (hunk 3 above)
Comment (dorschw) [reply pyproject.toml:16]: @ilaner do we **need** it?

**Review by dorschw — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -123,6 +124,7 @@ google-cloud-bigquery = "^3.2.0"
 db-dtypes = "^1.0.1"
 google-cloud-secret-manager = "^2.13.0"
 json5 = "^0.9.10"
+dateparser = "^1.1.8"
```
[the remark this answers is not in the document]
Comment (ilaner) [reply pyproject.toml:127]: let's add to the `dev` group as it is a base dependency

File: pyproject.toml (hunk 3 above)
Comment (ilaner) [reply pyproject.toml:16]: it will work on both cases, as `flake8` is installed as a part of demisto-sdk.
I think we can add this as dependency to the content, as we use it in content as well (if we someday plan to remove `flake8` from demisto-sdk, it still should remain on content.

Comment (dorschw) [reply pyproject.toml:16]: As none of the tasks here explicitly call flake8, please remove it.

File: poetry.lock [hunk 6]
```diff
@@ -141,17 +141,18 @@ frozenlist = ">=1.1.0"
 
 [[package]]
 name = "ansible-runner"
-version = "2.3.2"
+version = "2.3.3"
```
Comment (dorschw) [inline poetry.lock:144]: Are these changes intended? 
If not, run `poetry lock --no-update`

Comment (merit-maita) [reply poetry.lock:144]: no, there're not, will do now

**Comment by merit-maita** [comment]
I've tested running all changed github actions:

1. [Contribution form filled](ttps://github.com/demisto/content/actions/runs/5344677427/jobs/9689347992?pr=27654)
2. [Create internal pr from external](https://github.com/demisto/content/actions/runs/5344777220/jobs/9689578234?pr=27654)
3. [Handle new external pr](https://github.com/demisto/content/actions/runs/5344923125/jobs/9689903723)
4. [Handle stale branches](https://github.com/demisto/content/actions/runs/5345160971/jobs/9690460683?pr=27435)
5. [Sync contribution base branch](https://github.com/demisto/content/actions/runs/5345253899/jobs/9690683823?pr=27435)
6. [Sync contribution base branch on changes](https://github.com/demisto/content/actions/runs/5345375700/jobs/9690970236?pr=27435)

@dorschw @DeanArbel FYI

**Review by dorschw — APPROVED** [review]
